content/java/concepts/comparable/comparable.md

```diff
@@ -0,0 +1,106 @@
+---
+Title: 'Comparable'
+Description: 'The Comparable interface is used to define how a class is to be sorted.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Collections'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`Comparable`** interface is used to define how a [class](https://www.codecademy.com/resources/docs/java/classes) is to be sorted. It is not to be confused with the [`Comparator`](https://www.codecademy.com/resources/docs/java/comparator) interface, which is implemented in a separate class. The `Comparable` interface is implemented in the class to be sorted.
+
+The `Comparable` interface only has one method, `.compareTo()` that is used to compare the current instance of the class with another instance of the same class, and the sort is based on the return value of `.compareTo()`:
+
+| Return Value | Means                           |
+| :----------: | ------------------------------- |
+|    `>= 1`    | this instance > passed instance |
+|     `0`      | this instance = passed instance |
+|   `<= -1`    | this instance < passed instance |
+
+This method determines how items are sorted by methods such as [`Arrays.sort()`](https://www.codecademy.com/resources/docs/java/arrays/sort) and `Collections.sort()`.
+
+## Syntax
+
+```pseudo
+class MyClass implements Comparable<MyClass> {
+  // Class body.
+  ...
+  @Override public int compareTo(MyClass value)
+  {
+      // Comparison Logic
+      ...
+      return result;
+  }
+}
+```
+
+Using the interface in `MyClass` requires it to implement `Comparable<MyClass>` and override the `.compareTo()` method. The `.compareTo()` method can then use any applicable logic to decide if `this` instance comes before, after, or in the same place as `value` in the sort order and return the appropriate `int` value.
+
+## Example
+
+The following example defines an `Employee` class that sorts based on `lastName` then on `firstName`.
+
+```java
+import java.io.*;
```

Comment:
Since this example is a little long it might be a good idea to add comments just to explain what is going on.